<?xml version="1.0"?>
<launch>
  <arg name="mokeup_pose" default="false" />
  <!-- pick object server required components, moveit must be launched separately -->

  <!-- pick object state machine, wrapped around action lib -->
  <group if="$(arg mokeup_pose)">
    <node pkg="mir_pick_object" type="pick_object_server_whole_body_control.py" name="wbc_pick_object_server" output="screen" args="mokeup"/>
  </group>
  <group unless="$(arg mokeup_pose)">
    <!-- object selector -->
    <include file="$(find mcr_perception_selectors)/ros/launch/object_selector.launch" />
    
    <node pkg="mir_pick_object" type="pick_object_server_whole_body_control.py" name="wbc_pick_object_server" output="screen"/>
  </group>

  <!-- whole body motion control to grasp objects -->
  <include file="$(find mir_pick_object)/ros/launch/demo_youbot_whole_body_control.launch">
      <arg name="object_pose_topic" value="/mcr_perception/object_selector/output/object_pose" />
      <arg name="use_mokeup_pose" value="$(arg mokeup_pose)" />
  </include>

  <!-- dynamic reconfigure from code, used to configure pregrasp parameters during runtime -->
  <!--include file="$(find mcr_dynamic_reconfigure_client)/ros/launch/dynamic_reconfigure_client.launch">
    <arg name="config_file" value="$(find mir_yb_actions)/ros/config/yb_manipulation_params.yaml" />
  </include-->

</launch>
